app/jobs/laa_reference_updater_job.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class LaaReferenceUpdaterJob < ApplicationJob
+  queue_as :default
+
+  def perform(*args)
+    Api::RecordLaaReference.new(*args).call
```

Comment:
perhaps
```
MakeCommonPlatformCalls.call...
```

Reply:
Do you mean add a `MakeCommonPlatformCalls` service class, and call that from the job?

So I could move the logic from the `make_common_platform_calls` method in `LaaReferencesController` to that new service class which would do the looping and make the requests, and call that from the `LaaReferenceUpdaterJob`, which would be called from `LaaReferencesController`.

I think I was considering it slightly differently, where each request to CP would be a new job - so each request is made asynchronously rather than having a single async job making a number of requests.

So I alternatively I could move the logic from the `make_common_platform_calls` method in `LaaReferencesController` to a new service, which would be called from `LaaReferencesController` and would loop through the offences, but would create a new `LaaReferenceUpdaterJob` for each offence to actually make the call to CP.

But I'm not sure I have a strong opinion on it either way, if you can see a drawback to my approach I'm happy to go with what you suggest.